gobblin-runtime/src/main/java/gobblin/runtime/job_catalog/FSJobCatalog.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.runtime.job_catalog;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Properties;
+
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.typesafe.config.Config;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.runtime.api.JobSpec;
+import gobblin.runtime.api.JobSpecNotFoundException;
+import gobblin.runtime.api.MutableJobCatalog;
+import gobblin.runtime.util.FSJobCatalogHelper;
+import gobblin.util.ConfigUtils;
+import gobblin.util.PathUtils;
+import gobblin.util.filesystem.PathAlterationDetector;
+import gobblin.util.filesystem.PathAlterationListenerAdaptor;
+import gobblin.util.filesystem.PathAlterationObserver;
+
+
+/**
+ * The job Catalog for file system to persist the job configuration information.
+ * This implementation has no support for caching.
+ */
+public class FSJobCatalog extends ImmutableFSJobCatalog implements MutableJobCatalog {
+
+  /**
+   * Listeners of JobCatalog, Like GobblinInstaceDriver.
+   * Distinguished it from JobSpec Listeners.
+   */
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FSJobCatalog.class);
+
+  // A monitor for changes to job conf files for general FS
+  // This embedded monitor is monitoring job configuration files instead of JobSpec Object.
+  private final PathAlterationDetector pathAlterationDetector;
+
+  private final PathAlterationObserver observer;
+
+  /**
+   * Initialize the JobCatalog, fetch all jobs in jobConfDirPath.
+   * @param sysConfig
+   * @throws Exception
+   */
+  public FSJobCatalog(Config sysConfig)
+      throws Exception {
+    this(sysConfig, null);
```

Comment:
This null issue is addressed by `addPathAlterationObserver` method, which contains `Optional<PathAlterationObserver>` as a parameter. 

The initialization of this parameter is: `Optional<PathAlterationObserver> observerOptional = Optional.fromNullable(this.observer);` 

It could be null here and leave it to the lower layer methods to handle.